src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -194,6 +214,92 @@ MultiLogicalPlanCreate(Query *originalQuery, Query *queryTree,
 
 
 /*
+ * ReplaceExternParamsOnOriginalQuery replaces the external parameters that appears
+ * on the query with the corresponding entries on the boundParams.
+ *
+ * Note that this function is inspired by eval_const_expr() on Postgres. We failed to
+ * use that function given that it requires access to PlannerInfo.
+ */
+static Node *
+ReplaceExternParamsOnOriginalQuery(Node *inputNode, ParamListInfo boundParams)
```

Comment:
Makes sense, though it kind of sucks to have 4 (!) different ways of handling prepared statements though: deferred error, `ReplaceExternParamsOnOriginalQuery`, `ExecuteMasterEvaluableFunctions`, send parameters in router executor.

I think we need to do some significant refactoring after 6.2 to make the planner accessible for future generations. We should generally decouple planning from pruning.